table/data_block_hash_index.cc

```diff
@@ -12,115 +12,78 @@
 
 namespace rocksdb {
 
-const uint32_t kSeed = 2018;
-const uint32_t kSeed_tag = 214; /* second hash seed */
-
-inline uint16_t HashToBucket(const Slice& s, uint16_t num_buckets) {
-  return static_cast<uint16_t>(
-      rocksdb::Hash(s.data(), s.size(), kSeed) % num_buckets);
-}
-
 void DataBlockHashIndexBuilder::Add(const Slice& key,
-                                    const uint16_t& restart_index) {
-  uint16_t idx = HashToBucket(key, num_buckets_);
-  /* push a TAG to avoid false postive */
-  /* the TAG is the hash function value of another seed */
-  uint16_t tag = static_cast<uint16_t>(
-      rocksdb::Hash(key.data(), key.size(), kSeed_tag));
-  buckets_[idx].push_back(tag);
-  buckets_[idx].push_back(restart_index);
-  estimate_ += 2 * sizeof(uint16_t);
+                                    const uint8_t& restart_index) {
+  assert(Valid());
+  // TODO(fwu): error handling
```

Comment:
When the restart_index is too large, i.e. > 253, we should gracefully fall back to Binary Index instead of a assert() statement.

My plan is to invalidate the hash index builder when num_restarts > 253.